tox_extension_messages.h

```diff
@@ -4,6 +4,8 @@
 
 struct ToxExtensionMessages;
 
+#define TOX_EXTENSION_MESSAGES_MAX_MESSAGE_SIZE 10*1024*1024
```

Comment:
Even though this is a generous limit, I'm quite certain if we hardcode this limit here, we'll sooner or later run into problems again.

I'd prefer to make this configurable for the extension and use your value as default.

Reply:
Yeah, agreed that it would be better for it to be configurable. I'll add that.

Reply:
Hmm.. configuring the receive side buffer could cause some bad user experience, e.g. if the two friends have different configured buffers, friend A with size 4 and friend B with size 5, B might keep sending 5-length messages to A who silently drops them.

It seems like each friend should probably know the receive buffer size of their friend and avoid sending over that, which would require some additional negotiation. That size would then need to be exposed up to the user of the client, per-friend, so that they can appropriately limit message size per friend.

Reply:
To summarize chats with @sudden6 and @sphaerophoria, @sudden6 favours having a constant size and if the size needs to be increased in the future, a extension with a new UUID would be released with the larger constant size. A client could then support both extensions for backwards compatibility, assuming the second extension changed all symbol names as well to avoid conflicts, with the main argument being simplicity of extension implementation and of client implementation (which wouldn't need a different max size for each friend), since increasing the size may either never be needed or not be needed for many years.

@sphaerophoria favours adding the negotiation of receive buffer size per friend, since it would remove the need for a full copy of the extension to change the size, allows more customization per client, and a cleaner upgrade path, and is more conceptually "correct".

Given that this is a tox related project, idealistic solutions seem fitting. I think adding the negotiation wouldn't be too tough, on the toxext callback the extension can then do its own negotiation, and once that's complete call the client callback with the max size, which doesn't seem too complicated.

Thinking about the upgrade path though, if any extension protocol change other than size is needed, in general extensions couldn't do anything clever to support backwards compatibility since they would have a new UUID, and toxext would completely separate them. For e.g. messages, that would mean max size would drop back down to ~1KB and maybe the client could nag the user to nag their friend to update their extension, or the client could link both the old and new extensions then do backwards compatibility itself.. I wonder if there's much to be gained from allowing extensions to version themselves. I guess any extension can always add its own protocol version and use that rather than making a new UUID on protocol changes, so toxext wouldn't need to fundamentally support it. @sphaerophoria what do you think about adding a protocol version to `messenges`? Should that just be handled by either other extensions that wrap it, or by full UUID changes instead?